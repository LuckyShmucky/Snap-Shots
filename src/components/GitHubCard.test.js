import { render, screen, waitFor } from '@testing-library/react';
import GitHubCard from './GitHubCard';
import renderer  from "react-test-renderer"


beforeEach(()=>{
    fetch.resetMocks()
})
it("should have a src from github api for image", async ()=>{
    fetch.mockResponseOnce(JSON.stringify({avatar_url: 'https://avatars.githubusercontent.com/u/92145822?v=4'}))
    render(<GitHubCard/>)
    const cardImage = await screen.findByAltText('profile picture')
    expect(cardImage).toHaveAttribute('src', expect.stringContaining('avatars.githubusercontent.com'))
})


// it('should have a placeholder src for image', ()=>{
   
//     render(<GitHubCard/>)
//     const cardImage  = screen.getByRole('cardImage')
//     expect(cardImage).toHaveAttribute('src', expect.stringContaining('holder.js/100px180'))
// })


test('renders snapshot of github card', ()=>{
    const tree = renderer.create(<GitHubCard />).toJSON()
    expect(tree).toMatchSnapshot()
})