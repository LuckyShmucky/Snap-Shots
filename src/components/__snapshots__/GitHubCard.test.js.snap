// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot of github card 1`] = `
<div
  className="card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <img
    className="card-img-top"
    role="cardImage"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Card Title
    </div>
    <p
      className="card-text"
    >
      Some quick example text to build on the card title and make up the bulk of the card's content.
    </p>
  </div>
</div>
`;
